pkg/chains/signing/x509/x509.go

```diff
@@ -37,18 +41,70 @@ type Signer struct {
 }
 
 // NewSigner returns a configured Signer
-func NewSigner(secretPath string, logger *zap.SugaredLogger) (*Signer, error) {
-	privateKeyPath := filepath.Join(secretPath, "x509.pem")
-	if contents, err := ioutil.ReadFile(privateKeyPath); err == nil {
-		return x509Signer(contents, logger)
-	}
+func NewSigner(secretPath string, useSSL bool, logger *zap.SugaredLogger) (*Signer, error) {
+
+	x509PrivateKeyPath := filepath.Join(secretPath, "x509.pem")
+	cosignPrivateKeypath := filepath.Join(secretPath, "cosign.key")
 
-	privateKeyPath = filepath.Join(secretPath, "cosign.key")
-	if contents, err := ioutil.ReadFile(privateKeyPath); err == nil {
-		return cosignSigner(secretPath, contents, logger)
+	var signer *Signer
+	if contents, err := ioutil.ReadFile(x509PrivateKeyPath); err == nil {
+		signer, err = x509Signer(contents, logger)
+		if err != nil {
+			return nil, err
+		}
+	} else if contents, err := ioutil.ReadFile(cosignPrivateKeypath); err == nil {
+		signer, err = cosignSigner(secretPath, contents, logger)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		return nil, errors.New("no valid private key found, looked for: [x509.pem, cosign.key]")
 	}
+	if useSSL {
+		s, err := ssl.NewEnvelopeSigner(&sslAdapter{
+			s: signer,
+		})
+		if err != nil {
+			return nil, err
+		}
+		return &Signer{Signer: &sslSigner{s: s}}, nil
+	}
+	return signer, nil
+
+}
+
+type sslAdapter struct {
+	s     *Signer
+	keyID string
+}
+
+func (w *sslAdapter) Sign(data []byte) ([]byte, string, error) {
+	sig, _, err := w.s.Sign(context.Background(), data)
+	return sig, w.keyID, err
+}
 
-	return nil, errors.New("no valid private key found, looked for: [x509.pem, cosign.key]")
+func (w *sslAdapter) Verify(keyID string, data, sig []byte) (bool, error) {
+	panic("unimplemented")
+}
+
+type sslSigner struct {
+	s *ssl.EnvelopeSigner
+}
+
+func (s *sslSigner) PublicKey(ctx context.Context) (crypto.PublicKey, error) {
+	panic("unimplemented")
+}
+
+func (s *sslSigner) Sign(ctx context.Context, payload []byte) ([]byte, []byte, error) {
+	env, err := s.s.SignPayload("application/vnd.in-toto+json", payload)
```

Comment:
The payload type is defined in in-toto here: https://github.com/in-toto/in-toto-golang/blob/ite-6/in_toto/model.go#L45 to avoid raw strings.